p4_constraints/ast.cc

```diff
@@ -117,6 +118,16 @@ absl::optional<int> TypeBitwidth(const Type& type) {
   }
 }
 
+absl::StatusOr<int> TypeBitwidthOrStatus(const Type& type) {
+  std::optional<int> bitwidth = TypeBitwidth(type);
+  if (bitwidth.has_value()) {
+    return *bitwidth;
+  } else {
+    return gutils::InvalidArgumentErrorBuilder()
+           << "expected a type with bitwidth, but got: " << type;
```

Comment:
You'd probably have to bump up the protobuf version for <osstream> << <proto> to be supported.

Alternatively, you can just use `<< proto.DebugString()` or similar.

Reply:
Seems to work.